packages/syft/src/syft/core/tensor/autodp/single_entity_phi.py

```diff
@@ -959,6 +959,62 @@ def clip(
             scalar_manager=self.scalar_manager,
         )
 
+    def copy(
+        self, order: Optional[str] = "K", subok: Optional[bool] = True
+    ) -> SingleEntityPhiTensor:
+        """Return copy of the given object"""
+        data = np.array(self.child, order=order, subok=subok, copy=True)
+        min_vals = np.array(self.min_vals, order=order, subok=subok, copy=True)
+        max_vals = np.array(self.max_vals, order=order, subok=subok, copy=True)
+
+        return SingleEntityPhiTensor(
+            child=data,
+            entity=self.entity,
+            min_vals=min_vals,
+            max_vals=max_vals,
+            scalar_manager=self.scalar_manager,
+        )
+
+    def take(
+        self,
+        indices: np.ArrayLike,
+        axis: Optional[int] = None,
+        mode: Optional[str] = "raise",
+    ) -> SingleEntityPhiTensor:
+        """Take elements from an array along an axis"""
+        data = np.take(self.child, indices=indices, mode=mode)
+        return SingleEntityPhiTensor(
+            child=data,
+            entity=self.entity,
+            min_vals=self.min_vals,
```

Comment:
Seems like this would lead to a tensor whose min_vals have a different shape than the data since min_vals wasn't transformed in line with take. Let's run self.min_vals.take() as well for min_vals and max_vals